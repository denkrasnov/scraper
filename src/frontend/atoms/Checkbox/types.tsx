import { ReactElement } from "react";

type StyledColor = "TV8" | "NTV" | "JurnalTV";

export interface CheckboxProps {
  name: string;
  value: string;
  label?: string | ReactElement<any>;
  styled?: StyledColor;
  onChange: (event: any) => void;
}
