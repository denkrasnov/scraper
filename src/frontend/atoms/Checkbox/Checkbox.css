.checkbox {
  display: inline-flex;
  position: relative;
  align-items: center;
}

.checkbox > div {
  margin-left: 10px;
}

.checkbox > input {
  width: 20px;
  height: 20px;
  appearance: none;
  border: 1px solid var(--border-gray);
  border-radius: 4px;
  outline: none;
  transition: background 175ms cubic-bezier(0.1, 0.1, 0.25, 1);
  position: relative;
}

.checkbox > input:checked {
  border: 1px solid var(--main-red);
  background-color: var(--main-red);
}

.checkbox > input::before {
  content: "";
  display: block;
  position: absolute;
  top: 1px;
  left: 6px;
  width: 7px;
  height: 12px;
  border: solid var(--white);
  border-width: 0 2px 2px 0;
  transform: rotate(45deg);
  opacity: 0;
}

.checkbox > input:checked::before {
  opacity: 1;
}
