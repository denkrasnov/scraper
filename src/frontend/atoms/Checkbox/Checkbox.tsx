import React, { FC } from "react";

import { CheckboxProps } from "./types";
import styles from "./Checkbox.css";
import Box from "../Box";

const Checkbox: FC<CheckboxProps> = (props) => {
  const { name, value, label, onChange } = props;

  return (
    <label className={styles.checkbox}>
      <input name={name} onChange={onChange} type="checkbox" value={value} />
      <Box alignItems="center">{label}</Box>
    </label>
  );
};

export default Checkbox;
