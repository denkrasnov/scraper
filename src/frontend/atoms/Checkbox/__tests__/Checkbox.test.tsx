import { shallow } from "enzyme";

import Checkbox from "..";

describe("Checkbox", () => {
  it("should render", () => {
    const component = shallow(
      <Checkbox name="__TEST__" onChange={jest.fn()} value="__VALUE__" />
    );

    expect(component).toMatchSnapshot();
  });

  it("should render with label", () => {
    const component = shallow(
      <Checkbox
        label="__LABEL__"
        name="__TEST__"
        onChange={jest.fn()}
        value="__TEST__"
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("should render styled", () => {
    const component = shallow(
      <Checkbox
        label="__LABEL__"
        name="__TEST__"
        onChange={jest.fn()}
        styled="TV8"
        value="__TEST__"
      />
    );

    expect(component).toMatchSnapshot();
  });
});
