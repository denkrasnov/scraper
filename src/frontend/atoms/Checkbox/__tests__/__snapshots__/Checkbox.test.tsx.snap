// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render 1`] = `
<label
  className="root"
>
  <input
    name="__TEST__"
    onChange={[MockFunction]}
    type="checkbox"
    value="__VALUE__"
  />
  <Box
    alignItems="center"
  />
</label>
`;

exports[`Checkbox should render styled 1`] = `
<label
  className="root styled-TV8"
>
  <input
    name="__TEST__"
    onChange={[MockFunction]}
    type="checkbox"
    value="__TEST__"
  />
  <Box
    alignItems="center"
  >
    __LABEL__
  </Box>
</label>
`;

exports[`Checkbox should render with label 1`] = `
<label
  className="root"
>
  <input
    name="__TEST__"
    onChange={[MockFunction]}
    type="checkbox"
    value="__TEST__"
  />
  <Box
    alignItems="center"
  >
    __LABEL__
  </Box>
</label>
`;
